File: src/test/java/no/nav/tjenestepensjon/simulering/config/ExecutorServiceConfigTest.java
package no.nav.tjenestepensjon.simulering.config;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.core.Is.is;
import static org.mockito.Mockito.when;

import static no.nav.tjenestepensjon.simulering.domain.TpLeverandor.EndpointImpl.SOAP;

import java.util.List;
import java.util.concurrent.ThreadPoolExecutor;

import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.mockito.Mock;
import org.mockito.junit.jupiter.MockitoExtension;

import no.nav.tjenestepensjon.simulering.domain.TpLeverandor;

@ExtendWith(MockitoExtension.class)
class ExecutorServiceConfigTest {

    @Mock
    private TpLeverandorConfig tpLeverandorConfig;
    private ExecutorServiceConfig executorServiceConfig = new ExecutorServiceConfig();

    @Test
    void createOneThreadPerProvider() {
        when(tpLeverandorConfig.tpLeverandorList()).thenReturn(List.of(new TpLeverandor("lev1", "url1", SOAP), new TpLeverandor("lev2", "url2", SOAP)));

        ThreadPoolExecutor executorService = (ThreadPoolExecutor) executorServiceConfig.taskExecutor(tpLeverandorConfig);
        assertThat(executorService.getCorePoolSize(), is(2));
    }
}

File: src/test/kotlin/no/nav/tjenestepensjon/simulering/config/ExecutorServiceConfigTest.kt
package no.nav.tjenestepensjon.simulering.config

import io.mockk.every
import io.mockk.impl.annotations.MockK
import io.mockk.junit5.MockKExtension
import no.nav.tjenestepensjon.simulering.model.domain.TpLeverandor
import no.nav.tjenestepensjon.simulering.model.domain.TpLeverandor.EndpointImpl.SOAP
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith
import java.util.concurrent.ThreadPoolExecutor

@ExtendWith(MockKExtension::class)
internal class ExecutorServiceConfigTest {

    @MockK
    private lateinit var tpLeverandorConfig: TpLeverandorConfig

    private val executorServiceConfig: ExecutorServiceConfig = ExecutorServiceConfig()

    @Test
    fun `Create one thread per provider`() {
        every { tpLeverandorConfig.tpLeverandorList() } returns listOf(
            TpLeverandor("lev1", SOAP, "sim2", "stilling2"),
            TpLeverandor("lev2", SOAP, "sim2", "stilling2")
        )
        val executorService = executorServiceConfig.taskExecutor(tpLeverandorConfig) as ThreadPoolExecutor
        assertEquals(2, executorService.corePoolSize)
    }
}
